pkg/agent/inteface.go

```diff
@@ -112,18 +114,31 @@ const (
 
 	// InstallByLabel indicate to install addon based on clusters' label
 	InstallByLabel StrategyType = "LabelSelector"
+
+	// InstallByFilterFunction indicate to install addon based on a filter function
+	// If filter function returns true or it's nil, the addon will be installed to the cluster
```

Comment:
will we install the addon when the function is nil?

Reply:
Yes， if we set nil, in the construction function it will set a `filterfunction` always return true.

Reply:
There is no-way for framework users to set the function to `nil`, because we `managedclusterfilter` is a inline-struct only provide `Get` outside of pkg.